python/llm/src/bigdl/llm/transformers/speculative.py

```diff
@@ -430,14 +457,17 @@ def speculative_generate(self,
 
             if self.device.type == 'cpu':
                 # init past_key_values_storage and assign initial fp32 value
-                if step == 1:
+                if step == 1 and self.config.model_type != "gpt_bigcode":
                     past_key_values_storage = \
-                        _prepare_past_key_values_storage_cpu(self, past_key_values,
-                                                             max_new_tokens, _enable_ipex)
+                            _prepare_past_key_values_storage_cpu(self, past_key_values,
+                                                                 max_new_tokens, _enable_ipex)
                 # each iter cut off cur_len kv_cache from past_key_values1
-                draft_past_key_values = \
-                    _prepare_draft_past_key_values_cpu(self, past_key_values,
-                                                       past_key_values_storage)
+                if self.config.model_type == "gpt_bigcode":
```

Comment:
instead of checking `self.config.model_type == "gpt_bigcode"` here,  have `_prepare_draft_past_key_values_cpu` to correctly handle `gpt_bigcode` instead